### PR TITLE
Parse protocol version

### PR DESCRIPTION
**Pull request by mtymek**
As requested in #120.

### CODE REVIEW

**Comment by akrabat** [comment]
Looks good to me.

File: src/ServerRequestFactory.php [hunk 1]
```diff
@@ -70,6 +71,7 @@ public static function fromGlobals(
         );
 
         return $request
+            ->withProtocolVersion(static::marshalProtocolVersion($_SERVER))
```
Comment (snapshotpl) [inline src/ServerRequestFactory.php]: I think that you should use here `$server`

File: src/ServerRequestFactory.php [hunk 2]
```diff
@@ -70,6 +71,7 @@ public static function fromGlobals(
...
             ->withCookieParams($cookies ?: $_COOKIE)
```
Comment (snapshotpl) [inline src/ServerRequestFactory.php]: I'm little worried about this. Why we need to create 5 instances of `ServerRequest`? `ServerRequest` should be able to be constructed with all possible args. Maybe we can named constructor here?

Comment (snapshotpl) [inline src/ServerRequestFactory.php]: I prepared patch for that #124

File: src/ServerRequestFactory.php (hunk 1 above)
Comment (weierophinney) [inline src/ServerRequestFactory.php]: Agreed; I'll make that change on merge.

File: src/ServerRequestFactory.php [hunk 3]
```diff
@@ -455,4 +457,23 @@ private static function normalizeNestedFileSpec(array $files = [])
...
         return $normalizedFiles;
     }
+
+    /**
+     * Return HTTP protocol version (X.Y)
+     *
+     * @param array $server
+     * @return string
+     */
+    public static function marshalProtocolVersion(array $server)
```
Comment (weierophinney) [inline src/ServerRequestFactory.php]: This should also be private, for consistency with the other methods. I'll make that change on merge.

**Pull request merged**